packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -596,16 +608,20 @@ class FloatingActionButton extends StatelessWidget {
       case _FloatingActionButtonType.extended:
         sizeConstraints = floatingActionButtonTheme.extendedSizeConstraints ?? _kExtendedSizeConstraints;
         final double iconLabelSpacing = extendedIconLabelSpacing ?? floatingActionButtonTheme.extendedIconLabelSpacing ?? 8.0;
-        const Widget width20 = SizedBox(width: 20.0);
-        const Widget width16 = SizedBox(width: 16.0);
+        final EdgeInsetsGeometry padding = extendedPadding
+            ?? floatingActionButtonTheme.extendedPadding
+            ?? (child != null && isExtended ? const EdgeInsetsDirectional.only(start: 16.0, end: 20.0) : const EdgeInsetsDirectional.only(start: 20.0, end: 20.0));
         resolvedChild = _ChildOverflowBox(
-          child: Row(
-              mainAxisSize: MainAxisSize.min,
-              children: child == null
-                  ? <Widget>[width20, _extendedLabel!, width20]
-                  : isExtended
-                      ? <Widget>[width16, child!, SizedBox(width: iconLabelSpacing), _extendedLabel!, width20]
-                      : <Widget>[width20, child!, width20],
+          child: Padding(
+            padding: padding,
+            child: Row(
+                mainAxisSize: MainAxisSize.min,
+                children: child == null
```

Comment:
This can be reduced to an array with `if` conditionals inside

Reply:
Done